source/checkpoint_restart.f90

```diff
@@ -165,728 +248,349 @@ subroutine cr_killSwitch(iTurn)
+    call aper_crcheck_positionFiles
+  end if
 
-  !--   Just abort if we cannot re-position/copy the binary files,
-#ifndef STF
-102 continue
-  write(lerr,"(2(a,i0))") "SIXTRACR> ERROR PROBLEMS RE-READING fort.",myia," IOSTAT=",ierro
-  write(lerr,"(3(a,i0))") "          Unit ",myia," mybinrecs=",mybinrecs," Expected crbinrecs=",crbinrecs(ia)
-  write(lerr,"(a)")       "SIXTRACR> CRCHECK failure positioning binary files"
-  call prror(-1)
-105 continue
-  write(lerr,"(2(a,i0))") "SIXTRACR> ERROR PROBLEMS COPYING fort.",myia," IOSTAT=",ierro
-  write(lerr,"(4(a,i0))") "          Unit ",myia," mybinrecs=",mybinrecs,&
-    " Expected crbinrecs=",crbinrecs(ia)," binrecs94=",binrecs94
-  write(lerr,"(a)")       "SIXTRACR> CRCHECK failure copying binary files"
-  call prror(-1)
-#else
-102 continue
-  write(lerr,"(2(a,i0))") "SIXTRACR> ERROR PROBLEMS RE-READING singletrackfile.dat for ia=",ia," IOSTAT=",ierro
-  write(lerr,"(2(a,i0))") "          mybinrecs=",mybinrecs," Expected crbinrecs=",crbinrecs(ia)
-  write(lerr,"(a)")       "SIXTRACR> CRCHECK failure positioning binary files"
-  call prror(-1)
-105 continue
-  write(lerr,"(2(a,i0))") "SIXTRACR> ERROR PROBLEMS COPYING particle pair ",ia," IOSTAT=",ierro," from/to singletrackfile.dat"
-  write(lerr,"(3(a,i0))") "          mybinrecs=",mybinrecs," Expected crbinrecs=",crbinrecs(ia)," binrecs94=",binrecs94
-  write(lerr,"(a)")       "SIXTRACR> CRCHECK failure copying binary files"
-  call prror(-1)
-#endif
-  ! We are not checkpointing or we have no checkpoints
-  ! or we have no readable checkpoint
-  ! If not checkpointing we can just give up on lout and use
-  ! fort.6. We don't need to count records at all
-605 continue
-  write(93,"(a,l1)") "SIXTRACR> CRCHECK no restart possible checkp=",checkp
-  flush(93)
-  if (.not.checkp) then
-    if (rerun) then
-      ! we nevertheless have an existing fort.6
-      ! we will just overwrite it for now and delete
-      ! 92 to avoid abend copying it again
-      write(93,"(a)") "SIXTRACR> CRCHECK overwriting fort.6"
-      flush(93)
-    end if
-    ! and just use fort.6 from now on
-    write(93,"(a)") "SIXTRACR> CRCHECK giving up on LOUT"
-    flush(93)
-    ! Copy the lout to fort.6 (the file, not output_unit)
-    ! It seems that FORTRAN will open the file automatically?
-    ! There are no open(unit=6) etc. calls anywhere...
-    rewind(lout)
-3   read(lout,'(a1024)',end=1,err=107,iostat=ierro) arecord
-    lstring=1024
-    do i=1024,2,-1
-      lstring=i
-      if (arecord(i:i).ne.' ')goto 2
-      lstring=lstring-1
-    enddo
-2   write(6,'(a)') arecord(1:lstring)
-    goto 3
-    ! Not a flush?
-1   endfile(6,iostat=ierro)
-    backspace(6,iostat=ierro)
-    ! This is not a FLUSH!
-    rewind(lout)
-    endfile(lout,iostat=ierro)
-    close(lout)
-    lout=6
-  endif
+  ! Set up flag for tracking routines to call CRSTART
+  cr_restart = .true.
+  write(lout,"(a)") "SIXTRACR> "//repeat("=",80)
+  write(lout,"(a)") "SIXTRACR>  Restarted"
```

Comment:
I would not here say "restarted", so far it's only "armed" the restart.

Reply:
Yeah, this was already copied to the end of crstart, and shouldn't even be here. Will delete.